src/Components/Counter/__tests__/Counter.test.js

```diff
@@ -0,0 +1,41 @@
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+import { Counter } from '../Counter';
+
+describe('Counter tests', () => {
+  const initialValue = 5;
+
+  test('initial value is rendered correctly', async () => {
+    // Arrange
+    // Act
+    render(<Counter initialValue={initialValue} />);
+
+    // Assert
+    expect(await screen.findByTestId('counter-div')).toHaveTextContent(initialValue);
+  });
+
+  test('increment button is handled correctly', async () => {
```

Comment:
```suggestion
  test('increments counter on increment button click', async () => {
```

or something like this.